core/server/models/offer.js

```diff
@@ -0,0 +1,53 @@
+const ghostBookshelf = require('./base');
+
+const Offer = ghostBookshelf.Model.extend({
+    tableName: 'offers',
+
+    async onSaving(model, _attr, options) {
+        ghostBookshelf.Model.prototype.onSaving.apply(this, arguments);
+
+        if (model.get('name')) {
+            model.set('name', model.get('name').trim());
+        }
+
+        if (model.hasChanged('slug') || !model.get('slug')) {
```

Comment:
I'm wondering if we need this logic too - I think this can be handled by Offers module if necessary too!

Reply:
🤔 Yeah this is basically a copy of old practice we have been following so far for slugs, we could move it to offers module tho which might be cleaner. We'll essentially just need to duplicate the logic in `generateSlug` plugin method, but might be a cleaner cut so on same page

Reply:
Trimmed down models now so we can move this logic to members codebase

Reply:
Cool - sounds good!